polygeist/tools/cgeist/Lib/ValueCategory.h

```diff
@@ -34,11 +35,14 @@ class ValueCategory {
 public:
   mlir::Value val;
   bool isReference;
+  /// Holds the index the lvalue to a vector element refers to.
+  llvm::Optional<mlir::Value> Index;
```

Comment:
Should we initialize it to `llvm::None` by default? This would be initialized to an integer value only by one of the constructors (when val is a memref with a vector element type).

Reply:
Done